pkg/apis/monitoring/v1/thanos_types.go

```diff
@@ -68,6 +68,10 @@ type ThanosRulerSpec struct {
 	PodMetadata *EmbeddedObjectMetadata `json:"podMetadata,omitempty"`
 	// Thanos container image URL.
 	Image string `json:"image,omitempty"`
+	// Image pull policy for the 'thanos', 'init-config-reloader' and 'config-reloader' containers.
+	// See https://kubernetes.io/docs/concepts/containers/images/#image-pull-policy for more details.
+	// +kubebuilder:validation:Enum="";Always;Never;IfNotPresent
```

Comment:
Should we really allow users to set this to an empty string?
Also shouldn't we set a default? In my opinion it should be `Always`
```suggestion
	// +kubebuilder:validation:Enum=Always;Never;IfNotPresent
```

Reply:
Agree this should be set to `Always` by default, altho I believe in k8s `IfNotPresent` being default is to either maintain compatibility for air gapped environments or being able to run containers despite not being able to reach out remote registry.

Reply:
We want consistency with Kubernetes here. An empty `imagePullPolicy` doesn't necessarily default to `Always` (see https://kubernetes.io/docs/concepts/containers/images/#imagepullpolicy-defaulting) and I don't think we want to surprise people here.